### PR TITLE
Add "day zero" experiment feature & metrics

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves https://github.com/brave/brave-browser/issues/36243

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Only test for Windows and Android.

0. Start with fresh profile
1. Ensure `Brave.DayZero.Off.InstallTime` has a value of 0.
2. Close browser, advance system time by 7 days, start browser, ensure metric has value of 7.
3. Close browser, advance system time by 14 days, start browser with `--enable-features="BraveDayZeroExperiment"`, ensure metric has value of 21, and `Brave.DayZero.On.InstallTime` does NOT exist
4. Close browser advance system time by 14 days, start browser, ensure that the metric still has a value of 21 (to ensure that the metric was not updated internally).
5. Start with fresh profile with `--enable-features="BraveDayZeroExperiment"` flag
6. Ensure `Brave.DayZero.On.InstallTime` has a value of 0.
7. Close browser, advance system time by 7 days, start browser, ensure metric has value of 7.
8. Close browser, advance system time by 14 days, start browser without `--enable-features="BraveDayZeroExperiment"`, ensure metric has value of 21, and `Brave.DayZero.Off.InstallTime` does NOT exist
9. Close browser advance system time by 14 days, start browser, ensure that the metric still has a value of 21 (to ensure that the metric was not updated internally).

Ensure the metrics only exist in the `logs_constellation_prep_express` pref.

### CODE REVIEW

**Comment by DJAndries** [comment]
Will change base to `master` once https://github.com/brave/brave-core/pull/21583 lands.

File: android/java/org/chromium/base/BraveFeatureList.java [hunk 1]
```diff
@@ -31,6 +31,6 @@ public abstract class BraveFeatureList {
     public static final String AI_CHAT = "AIChat";
     public static final String BRAVE_SHOW_STRICT_FINGERPRINTING_MODE =
             "BraveShowStrictFingerprintingMode";
-    public static final String BRAVE_ZERO_DAY_FLAG_ANDROID = "BraveZeroDayFlagAndroid";
+    public static final String BRAVE_DAY_ZERO_EXPERIMENT = "BraveDayZeroExperiment";
```
Comment (chrismore) [inline android/java/org/chromium/base/BraveFeatureList.java:34]: @DJAndries the `BRAVE_DAY_ZERO_EXPERIMENT`variable is the same variable cross platforms and as a replacement for the previous Android-specific one?

File: components/misc_metrics/general_browser_usage.cc [hunk 2]
```diff
@@ -42,6 +58,20 @@ void GeneralBrowserUsage::ReportWeeklyUse() {
                              usage_storage_->GetLastISOWeekSum(), 8);
 }
 
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_WIN)
+void GeneralBrowserUsage::ReportInstallTime() {
+  int days_since_install = (base::Time::Now() - first_run_time_).InDays();
+  if (days_since_install < 0 || days_since_install > 30) {
```
Comment (chrismore) [inline components/misc_metrics/general_browser_usage.cc:64]: @DJAndries this condition is to just keep the histogram putting users into 0-30 day buckets and anyone outside of this range, gets returned to the previous function, right? e.g. negative or >=31 will not be included.

**Comment by chrismore** [comment]
@DJAndries If and when we want to bring this idea to Mac, will any of this approach work on Mac?

File: android/java/org/chromium/base/BraveFeatureList.java (hunk 1 above)
Comment (DJAndries) [reply android/java/org/chromium/base/BraveFeatureList.java:34]: yes, correct

File: components/misc_metrics/general_browser_usage.cc (hunk 2 above)
Comment (DJAndries) [reply components/misc_metrics/general_browser_usage.cc:64]: yes, that's right. users outside of the 0-30 day range will not report either metric.

**Comment by DJAndries** [comment]
> @DJAndries If and when we want to bring this idea to Mac, will any of this approach work on Mac?

Yes, once we add support for loading variations upon first run for Mac, it will work. We'll just need to make a couple tweaks.

**Review by iefremov — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.67.1 Chromium: 124.0.6367.29 (Official Build) nightly (64-bit)
-- | --
Revision | f4edcfc3cafa0d8609db03da09eb8582d83d1a27
OS | Windows 11 Version 23H2 (Build 22631.3374)
```

## `Test Case #1` - `--enable-features=BraveDayZeroExperiment:variant/a`

* launched `1.67.1 Chromium: 124.0.6367.29` using `--enable-features=BraveDayZeroExperiment:variant/a`
* re-launched once again using `--enable-features=BraveDayZeroExperiment:variant/a`
* ensured that `StudyBraveDayZeroExperiment:GroupBraveDayZeroExperiment` via `brave://version`

image

* ensured that `Brave.DayZero.A.InstallTime` was `0` via `brave://local-state`

```
            "Brave.DayZero.A.InstallTime": {
               "sent": false,
               "value": "0"
```

image

* closed the browser and moved the machines time/clock forwards by `7 days`
* re-launched once again using `--enable-features=BraveDayZeroExperiment:variant/a`
* ensured that `Brave.DayZero.A.InstallTime` was set as `7` via `brave://local-state`

```
            "Brave.DayZero.A.InstallTime": {
               "sent": false,
               "value": "7"
            },
```

image

* closed the browser and moved the machines time/clock forwards by `14 days`
* re-launched once again using `--enable-features=BraveDayZeroExperiment:variant/b`
* ensured that `Brave.DayZero.A.InstallTime` was set as `21` via `brave://local-state`
* ensured that `Brave.DayZero.B.InstallTim` doesn't exist under `brave://local-state`

```
            "Brave.DayZero.A.InstallTime": {
               "sent": false,
               "value": "21"
            },
```

**`Example`** | **`Example`**
----------------|-----------------
 | 

* closed the browser and moved the machines time/clock forwards by `14 days`
* re-launched once again using `--enable-features=BraveDayZeroExperiment:variant/b`
* ensured that `Brave.DayZero.A.InstallTime` was set as `21` via `brave://local-state`
* ensured that `Brave.DayZero.B.InstallTim` doesn't exist under `brave://local-state`

**Pull request merged**